### ISSUE TITLE
Skype for Business 2016: Installation causes Degredation of Output with NVDA on Windows 7

### ISSUE THREAD

**Issue by elliott94** [nvaccess/nvda#7311]
I realise that this is most likely not going to be easy to reproduce - I'll do my best to explain the issue.

After installing the Skype for Business 2016 client onto a Windows 7 Professional X64 machine, I've noticed that there is an immediate degredation of the output that NVDA provides. As an example, when opening the Start Menu, the word "list" is output before announcing the name of the item that has focus. Alt-tabbing only results in NVDA announcing "task switching"; the name of the window is not announced. When opening a new tab in Firefox, for example, no content at all can be read on the page. Explorer menus (including the Start Menu) don't output anything, and typing text into the Search Box results in the same. Uninstalling the client immediately restores output as it should be.

I'm not sure what the cause of this could be. I've done a complete reimage of the device in question; additionally, testing on another machine with the exact same Windows image (but different hardware) does not result in the same issue. I should also point out that the Skype client does not actually have to be running at all for this behaviour to occur - I've killed the process directly from within Task Manager and rebooted several times. At the moment, however, having Skype installed literally makes the machine unusable with NVDA. NVDA is running as an installed copy with full admin permissions.

**Comment by leonardder** [comment]
This might be a very strange question to you, but do you, by any chance, 
have Dolphin SuperNova installed on your system or has it ever been 
installed on the image you're using?

I recall the exact same problem on Windows 7 X86, but it occurred within 
Dolphin Screen Reader 15.05.

Anyhow, this is most certainly not related to NVDA itself. I rather 
think Skype for Business harms the system in some way.

**Comment by elliott94** [comment]
No; I've never installed anything other than NVDA.

Definitely worth considering, but the question would then be why it only happens with this one machine and not the other.

**Comment by leonardder** [comment]
What does NVDA do in the following cases?

1. Browsing the web with IE
2. Going to the task bar with Windows+t, and than using left/right arrow to navigate it.

**Comment by elliott94** [comment]
1. if in the Taskbar, NVDA will only output the names of tooltips for each icon (if there is an associated tooltip). Otherwise, nothing will be announced.

2. In IE, nothing whatsoever is announced when tabbing - other than the question of whether the browser should be set as the default.

**Comment by jcsteh** [comment]
This sounds like both UI Automation and MSAA are kinda broken, which is rather concerning. It'd be great if you could provide a complete log file at level debug showing a brief bit of alt+tabbing, navigation in the Start menu, tabbing in IE and moving through the icons on the Taskbar.

Also, you said:

> NVDA is running as an installed copy with full admin permissions.

Just to be super clear, do you mean you're choosing to run NVDA as admin or just that the account has admin privs? You should not run NVDA as admin (assuming you have UAC enabled).

**Comment by elliott94** [comment]
Section of a log attached - let me know if this is useful. The account that NVDA is running under has admin permissions - sorry for the lack of clarification!
[nvda.txt](https://github.com/nvaccess/nvda/files/1171068/nvda.txt)

**Comment by leonardder** [comment]
I've just received a message from someone who confirmed similar output degradation after installing SFB2016 on a Windows 7 X86 system. I've seen similar output degradation issues with SuperNova related to SFB2016 installation.

**Comment by jcsteh** [comment]
Can affected users please try these instructions I originally outlined in https://github.com/nvaccess/nvda/issues/2807#issuecomment-320149243:

> 1. Please apply [this registry patch](https://github.com/nvaccess/nvda/files/1199173/oleaccProxy.reg.txt) (remove the .txt extension).
> 2. If you're using 64 bit Windows, please also apply [this second registry patch](https://github.com/nvaccess/nvda/files/1199174/oleaccProxyWow64.reg.txt) (remove the .txt extension).
> 3. Open a command prompt as administrator: Start menu, type cmd.exe into the search box, press control+shift+enter and accept the UAC prompt.
> 4. Enter the following commands:
>     ```
>     c:\windows\system32\regsvr32 c:\windows\system32\oleaut32.dll
>     c:\windows\system32\regsvr32 c:\windows\system32\actxprxy.dll
>     ```
> 5. On a 64 bit system, also enter the following commands:
>     ```
>     c:\windows\syswow64\regsvr32 c:\windows\syswow64\oleaut32.dll
>     c:\windows\syswow64\regsvr32 c:\windows\syswow64\actxprxy.dll
>     ```
> 6. On Windows 7 (both 32 and 64 bit), also enter the following command:
>     ```
>     c:\windows\system32\regsvr32 "c:\Program Files\Internet Explorer\ieproxy.dll"
>     ```
> 7. On Windows 7 64 bit, also enter the following command:
>     ```
>     c:\windows\syswow64\regsvr32 "c:\Program Files (x86)\Internet Explorer\ieproxy.dll"
>     ```

**Comment by elliott94** [comment]
Yes; applying the above steps has now resolved this, and allowed NVDA to output content as expected.

**Comment by jcsteh** [comment]
Closing as a duplicate of #2807, since it's the same underlying cause and is fixable in the same way.
